_posts/2018-01-04-notification-channels-android-o.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title:  "Notification Channels in Android Oreo"
+date:   2018-02-06 20:20:02
+description: Prepare your app for Android Oreo targets by implementing notification channels
+excerpt: "Prepare your app for Android Oreo targets by implementing notification channels"
+permalink: /posts/notification-channels-android-o
+tags: [android oreo, tutorial, notification channels, notification categories]
+---
+
+
+In (not-so) recent news, Google is going to be enforcing that apps are targetted for SDK 26 (Android Oreo) by late 2018, in order to ensure security and improved performance in apps. You can read more about the announcement [here](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).
+
+Now, it's pretty easy to target API 26, but there's no gurantee that your app will function as it used to by just changing the `targetSdkVersion` value in your `build.gradle` file.
+
+This post looks specifically at one of the new things introduced in Android Oreo, and how targetting API 26 will affect your app, and notifications - notification channels (or categories).
+
+TL;DR
+-----
+
+Your app will stop sending notifications to the users if you target API 26 and you don't implement notification channels.
+
+What are Notification Channels?
+-----
+
+In Android O, notification channels are somewhat like groups or categories of notifications - say you're building a social networking app, the channels can be "activity" - like likes, or comments on your posts, "messages", etc. In fact, in the settings of your app, the user sees channels as "categories" as seen in the screenshot below.
+
+<!-- Insert notification settings screenshot here -->
+<p align="center">
+	<img src="/img/notification-channels-android-o.png"
+  alt="Android Studio showing app:fontFamily as error">
+</p>
+
+
+
+With the introduction of notification channels, users can get a fine control of what they want to be notified about - they can specifically turn off notifications for a certain channel, specify the importance as well as the preferred sound for a particular category of notifications, whether or not to override DND (Do not disturb)
```

Comment:
_users can get a fine control of what they want to be notified about - they can specifically_

I'd start a new sentence here. 

_users can get a fine control of what they want to be notified about. They can specifically_

-------------------------------------------------
_particular category of notifications, whether or not to override DND (Do not disturb)_

_particular category of notifications and **determine** whether or not to override DND (Do not disturb)**.**_

Reply:
🔒